app/controllers/revision_feedback_controller.rb

```diff
@@ -3,19 +3,54 @@
 require "#{Rails.root}/lib/importers/revision_score_importer"
 
 class RevisionFeedbackController < ApplicationController
+  before_action :initialize_ores
+
   def index
-    @wiki = Wiki.find_by(language: 'en', project: 'wikipedia')
-    @revision = Revision.find_by(mw_rev_id: params[:rev_id], wiki_id: @wiki.id)
-    set_new_revision unless @revision
-    @feedback = RevisionFeedbackService.new(@revision).feedback
+    set_revision_id
+    unless @revId.nil?
+      features = ores_features(@revId)
+      @feedback = RevisionFeedbackService.new(features).feedback
+    end
   end
 
   private
 
-  def set_new_revision
-    @revision = Revision.new(wiki_id: @wiki.id, mw_rev_id: params[:rev_id], mw_page_id: 0)
-    @revision.save!
-    RevisionScoreImporter.new.update_revision_scores([@revision])
-    @revision.reload
+  def initialize_ores
+    @wiki = Wiki.find_by(language: 'en', project: 'wikipedia')
+    @ores_api = OresApi.new(@wiki)
+  end
+
+    def ores_features(revId)
+    ores_data = @ores_api.get_revision_data(revId)
+    score = extract_score ores_data
+    return extract_features(ores_data)[revId.to_s]
+  end
+
+  def set_revision_id
+    @article = Article.find(params[:article_id])
+
+    query = { prop: 'revisions', pageids: @article.mw_page_id, rvprop: 'ids' }
+    response = WikiApi.new(@wiki).query(query)
+    results = response&.data
+    revisions = results.dig('pages', @article.mw_page_id.to_s, 'revisions')
+
+    # The API sends a response with the id of the last revision
+    unless revisions.nil? || revisions.length == 0
+      @revId = revisions[0]['revid']
```

Comment:
Standard Ruby style is to use snake_case for variable names: `@rev_id` in this case.